packages/unified-latex-ctan/package/minted/libs/command-argument-parser.ts

```diff
@@ -0,0 +1,45 @@
+import { arg } from "@unified-latex/unified-latex-builder";
+import { Argument, ArgumentParser } from "@unified-latex/unified-latex-types";
+import { parse as parseArgspec } from "@unified-latex/unified-latex-util-argspec";
+import { gobbleSingleArgument } from "@unified-latex/unified-latex-util-arguments";
+import { match } from "@unified-latex/unified-latex-util-match";
+
+/**
+ * This argument parser parses arguments in the form of
+ * - [⟨options⟩]{⟨language⟩}⟨delim⟩⟨code⟩⟨delim⟩
+ * - [⟨options⟩]{⟨language⟩}{⟨code⟩}
+ */
+export const argumentParser: ArgumentParser = (nodes, startPos) => {
+    const { argument: optionalArg, nodesRemoved: optionalArgNodesRemoved } =
+        gobbleSingleArgument(nodes, parseArgspec("o")[0], startPos);
+
+    const { argument: languageArg, nodesRemoved: languageArgNodesRemoved } =
+        gobbleSingleArgument(nodes, parseArgspec("m")[0], startPos);
+
+    let codeArg: Argument | null = null;
+    let codeArgNodesRemoved: number = 0;
+    const nextNode = nodes[startPos];
+    if (match.group(nextNode)) {
+        const mandatoryArg = gobbleSingleArgument(
+            nodes,
+            parseArgspec("m")[0],
```

Comment:
You can declare this as a `const` at the top of the file so you don't need to reparse the argspec on every call.